src/fft128/f128_impl.rs

```diff
@@ -0,0 +1,1075 @@
+        let s = simd.avx._mm256_add_pd(a, b);
+        (s, simd.avx._mm256_sub_pd(b, simd.avx._mm256_sub_pd(s, a)))
+    }
+
+    #[inline(always)]
+    pub(crate) fn _mm256_two_sum(simd: Avx, a: __m256d, b: __m256d) -> (__m256d, __m256d) {
+        let s = simd.avx._mm256_add_pd(a, b);
+        let bb = simd.avx._mm256_sub_pd(s, a);
+        (
+            s,
+            simd.avx._mm256_add_pd(
+                simd.avx._mm256_sub_pd(a, simd.avx._mm256_sub_pd(s, bb)),
+                simd.avx._mm256_sub_pd(b, bb),
+            ),
+        )
+    }
+
+    #[inline(always)]
+    pub(crate) fn _mm256_two_diff(simd: Avx, a: __m256d, b: __m256d) -> (__m256d, __m256d) {
+        let s = simd.avx._mm256_sub_pd(a, b);
+        let bb = simd.avx._mm256_sub_pd(s, a);
+        (
+            s,
+            simd.avx._mm256_sub_pd(
+                simd.avx._mm256_sub_pd(a, simd.avx._mm256_sub_pd(s, bb)),
+                simd.avx._mm256_add_pd(b, bb),
+            ),
+        )
+    }
+
+    #[inline(always)]
+    pub(crate) fn _mm256_two_prod(simd: Avx, a: __m256d, b: __m256d) -> (__m256d, __m256d) {
+        let p = simd.avx._mm256_mul_pd(a, b);
+        (p, simd.fma._mm256_fmsub_pd(a, b, p))
+    }
+
+    #[cfg(feature = "nightly")]
+    #[inline(always)]
+    pub(crate) fn _mm512_quick_two_sum(simd: Avx512, a: __m512d, b: __m512d) -> (__m512d, __m512d) {
+        let s = simd.avx512f._mm512_add_pd(a, b);
+        (
+            s,
+            simd.avx512f
+                ._mm512_sub_pd(b, simd.avx512f._mm512_sub_pd(s, a)),
+        )
+    }
+
+    #[cfg(feature = "nightly")]
+    #[inline(always)]
+    pub(crate) fn _mm512_two_sum(simd: Avx512, a: __m512d, b: __m512d) -> (__m512d, __m512d) {
+        let s = simd.avx512f._mm512_add_pd(a, b);
+        let bb = simd.avx512f._mm512_sub_pd(s, a);
+        (
+            s,
+            simd.avx512f._mm512_add_pd(
+                simd.avx512f
+                    ._mm512_sub_pd(a, simd.avx512f._mm512_sub_pd(s, bb)),
+                simd.avx512f._mm512_sub_pd(b, bb),
+            ),
+        )
+    }
+
+    #[cfg(feature = "nightly")]
+    #[inline(always)]
+    pub(crate) fn _mm512_two_diff(simd: Avx512, a: __m512d, b: __m512d) -> (__m512d, __m512d) {
+        let s = simd.avx512f._mm512_sub_pd(a, b);
+        let bb = simd.avx512f._mm512_sub_pd(s, a);
+        (
+            s,
+            simd.avx512f._mm512_sub_pd(
+                simd.avx512f
+                    ._mm512_sub_pd(a, simd.avx512f._mm512_sub_pd(s, bb)),
+                simd.avx512f._mm512_add_pd(b, bb),
+            ),
+        )
+    }
+
+    #[cfg(feature = "nightly")]
+    #[inline(always)]
+    pub(crate) fn _mm512_two_prod(simd: Avx512, a: __m512d, b: __m512d) -> (__m512d, __m512d) {
+        let p = simd.avx512f._mm512_mul_pd(a, b);
+        (p, simd.avx512f._mm512_fmsub_pd(a, b, p))
+    }
+
+    impl Avx {
+        #[inline(always)]
+        pub fn _mm256_add_estimate_f128_f128(
```

Comment:
as was discussed bench and compare the non estimate versions to be able to have the choice potentially for intrisics to select between estimate and more precise implementations